### ISSUE TITLE
Test MAE: 7.693

### ISSUE THREAD

**Issue by MaraniMatias** [deeplearningrosario/deeplearningforcomputervision#44]
# Run info and Model info
**Image layer:** Xception
**Image quantity:** 
- Males: 6780 (4520 training - 1130 validation - 1130 testing)
- Females: 5740 (3828 training - 956 validation - 956 testing)

**Epochs**: 100
**Batch size:** 32
**Loss**: Mean Squared Error
**Additional metrics:** Mean Absolute Error
**Optimizer:** Adam(lr=0.001, amsgrad=True)
**Callbacks:** 
- TrainingMonitor
- Checkpoint
- ReduceLROnPlateau`(monitor="val_loss", factor=0.8, patience=3, verbose=1, min_lr=0.0001)`
- CSVLogger

![summary_model](https://user-images.githubusercontent.com/7859628/45388230-c5ca1800-b5ee-11e8-9c8d-3085a61ac38f.png)

## Training
![history_loss](https://user-images.githubusercontent.com/7859628/45388288-e5f9d700-b5ee-11e8-9716-f3e645134f32.png)
![history_mean](https://user-images.githubusercontent.com/7859628/45388289-e5f9d700-b5ee-11e8-9c3c-551528670a68.png)

### For test data
**Test loss:** 103.37169375104163
**Test MAE:** 7.6935354787802765

![predicted_age_for_test_data_graph_1](https://user-images.githubusercontent.com/7859628/45408284-4b2ae800-b642-11e8-901a-ab84a56ef4a4.png)

### For validation data
![predicted_boone_age_for_valid_data_graph_1](https://user-images.githubusercontent.com/7859628/45410284-b4612a00-b647-11e8-945a-b89edf453e47.png)

### For training data
![predicted_boone_age_for_train_data_graph_1](https://user-images.githubusercontent.com/7859628/45410270-ad3a1c00-b647-11e8-9cbb-01ddfe3d02cb.png)

epoch | loss | lr | mean_absolute_error | val_loss | val_mean_absolute_error
-- | -- | -- | -- | -- | --
0 | 1451.72747723772 | 0.001 | 24.1343597682629 | 381.465455542589 | 15.1800753190213
1 | 294.421858417856 | 0.001 | 13.2935736701428 | 336.315794025834 | 14.4042695174396
2 | 225.493832977982 | 0.001 | 11.6505440450137 | 273.337783242918 | 13.3182236332221
3 | 173.49554533289 | 0.001 | 10.2828998894685 | 429.157444581775 | 16.4960926015676
4 | 149.140460728571 | 0.001 | 9.61991296776414 | 230.985738503739 | 12.4578878781254
5 | 135.898774439945 | 0.001 | 9.19890414029146 | 169.303337711616 | 10.124592773752
6 | 111.447338924947 | 0.001 | 8.34718772516219 | 154.949225048091 | 9.57771669161057
7 | 95.5547456873851 | 0.001 | 7.66065204777555 | 136.636422907975 | 8.6721159513479
8 | 81.7728699417736 | 0.001 | 7.13831983437422 | 152.929654568519 | 9.39290383449565
9 | 77.2069099912474 | 0.001 | 6.89252709177161 | 137.435230807399 | 8.65871011673661
10 | 69.3761097021976 | 0.001 | 6.62360193401532 | 222.874281256905 | 12.2196280825972
11 | 60.5686674351091 | 0.00080000004 | 6.11551034262035 | 151.657591895655 | 9.36867343887958
12 | 52.620523731504 | 0.00080000004 | 5.70277212153911 | 136.990978303143 | 8.88666399144715
13 | 54.345239608209 | 0.00080000004 | 5.78541784121994 | 125.058486345973 | 8.5440996920731
14 | 46.0875110087009 | 0.00080000004 | 5.33620789774194 | 144.239453916467 | 9.29955081903123
15 | 43.2094473041412 | 0.00080000004 | 5.15367288297024 | 127.740421401231 | 8.63471000375135
16 | 46.0967626164961 | 0.00080000004 | 5.29980300337605 | 181.766460063931 | 10.6415719729942
17 | 38.4211813328537 | 0.00064000004 | 4.8438330206165 | 113.988073508097 | 7.98986406042843
18 | 31.6843473839314 | 0.00064000004 | 4.42992033042231 | 146.721343899048 | 9.44778411568983
19 | 33.0979625316231 | 0.00064000004 | 4.53628720400947 | 117.691857861878 | 8.23278659370693
20 | 32.0481309785646 | 0.00064000004 | 4.42462315717038 | 117.053016252966 | 8.17547465559392
21 | 31.0513703767112 | 0.00051200006 | 4.33937919545414 | 116.169625030839 | 8.10700596647546
22 | 29.5626994170923 | 0.00051200006 | 4.27117050990684 | 113.927483991146 | 8.04409902687841
23 | 27.371115429069 | 0.00051200006 | 4.07333374240493 | 112.518280226798 | 8.00268164119922
24 | 25.296621669921 | 0.00051200006 | 3.90230143864188 | 115.983091114809 | 8.07416948490381
25 | 25.3259776151517 | 0.00051200006 | 3.90739169666855 | 114.750763392974 | 7.99370747536049
26 | 24.7632898569221 | 0.00051200006 | 3.87625769033464 | 117.413616023151 | 8.23911933450891
27 | 23.8452273351928 | 0.00040960003 | 3.7886785745735 | 115.642980431093 | 8.07747110264413
28 | 23.1106284812814 | 0.00040960003 | 3.74252289210888 | 110.327998866285 | 7.8832769499072
29 | 22.2372772160134 | 0.00040960003 | 3.63831000974662 | 111.438336980423 | 7.90455349431171
30 | 22.8725083885321 | 0.00040960003 | 3.69864851113536 | 110.92199263751 | 7.95574725142833
31 | 22.0647235353083 | 0.00040960003 | 3.67176413958759 | 117.925551128296 | 8.24163920767355
32 | 20.9908616660483 | 0.00032768003 | 3.5409472782645 | 112.125953209823 | 7.98263679323352
33 | 20.2874894674509 | 0.00032768003 | 3.49391093251805 | 112.136229470272 | 7.97570206814964
34 | 19.2084169981865 | 0.00032768003 | 3.38170578460346 | 112.347022187333 | 7.95203840058236
35 | 19.3185002775243 | 0.00026214402 | 3.42027548322509 | 113.323322475357 | 8.01033395957398
36 | 19.3922888194195 | 0.00026214402 | 3.38064927762356 | 114.530975700225 | 8.11143907627461
37 | 19.7623431480382 | 0.00026214402 | 3.43377387529363 | 109.951867477549 | 7.85353286611497
38 | 19.2828755111402 | 0.00026214402 | 3.38814017597888 | 109.323813305834 | 7.83814451587052
39 | 18.4315980715676 | 0.00026214402 | 3.33374447507269 | 109.944176557766 | 7.89429255848534
40 | 19.3051946002099 | 0.00026214402 | 3.3773504568163 | 109.654014214345 | 7.87512420670755
41 | 18.7600956598308 | 0.00026214402 | 3.32589519395175 | 110.971868966707 | 7.91119532571429
42 | 18.0792695937913 | 0.00020971522 | 3.28801534823613 | 110.278980631842 | 7.90337946728115
43 | 17.9372088236276 | 0.00020971522 | 3.27358774186095 | 111.746157700126 | 7.92239327170133
44 | 17.8249763201799 | 0.00020971522 | 3.272315969522 | 109.145740816257 | 7.87001622100347
45 | 17.2856495020546 | 0.00020971522 | 3.21860905977659 | 109.059793232729 | 7.82626925035954
46 | 17.7868996397639 | 0.00020971522 | 3.25784040993918 | 110.205456818739 | 7.88677077325399
47 | 18.0521588629379 | 0.00020971522 | 3.28226302489174 | 112.217939744403 | 8.00054062681482
48 | 17.7904485767403 | 0.00020971522 | 3.25724658945501 | 110.075368948888 | 7.83218286547199
49 | 17.0573058354552 | 0.00016777217 | 3.16516866556011 | 110.345953944088 | 7.92483738536231
50 | 17.2622446728255 | 0.00016777217 | 3.22006084316932 | 108.661857089283 | 7.79521317449992
51 | 16.5109587734718 | 0.00016777217 | 3.12323115736774 | 109.422452401795 | 7.83516336150275
52 | 17.3206577593479 | 0.00016777217 | 3.20415081381512 | 109.70189882902 | 7.84345587696577
53 | 16.8207861420979 | 0.00016777217 | 3.13784274565985 | 108.948656504586 | 7.80792831757382
54 | 16.985499889701 | 0.00013421774 | 3.17811452114359 | 108.831174793774 | 7.81494086495068
55 | 16.6197487804155 | 0.00013421774 | 3.12019089528346 | 111.157768666459 | 7.93605770795023
56 | 16.0672243289189 | 0.00013421774 | 3.0772549307512 | 110.50286872915 | 7.92712950912212
57 | 15.9097918710128 | 0.000107374195 | 3.06021737618257 | 110.334484246639 | 7.9259273939599
58 | 16.6960486643727 | 0.000107374195 | 3.11608513074454 | 108.920260786896 | 7.81752543709994
59 | 16.2940063321196 | 0.000107374195 | 3.10219398605007 | 109.411103728634 | 7.84149310412823
60 | 16.1216432400508 | 0.0001 | 3.07583611435335 | 110.478336977936 | 7.85756447079769
61 | 15.5932916102937 | 0.0001 | 3.03665607228025 | 110.06862203684 | 7.87945923800665
62 | 16.3442813496756 | 0.0001 | 3.10936789439521 | 109.790385022259 | 7.83466722455487
63 | 16.0498007256618 | 0.0001 | 3.07040596122541 | 108.729438946323 | 7.82340258034169
64 | 16.0982541868889 | 0.0001 | 3.07356997523287 | 109.89746044009 | 7.88653580591526
65 | 15.8669199566551 | 0.0001 | 3.07010858448487 | 108.409791615414 | 7.79925090903175
66 | 16.0972985768924 | 0.0001 | 3.08289099836327 | 109.233526107335 | 7.85055224573167
67 | 15.416142735118 | 0.0001 | 3.00987884309455 | 110.254448686784 | 7.86317988758691
68 | 15.8513014843205 | 0.0001 | 3.04467107670468 | 108.909685875517 | 7.83643445712608
69 | 16.17482851702 | 0.0001 | 3.05143120406516 | 109.280914924746 | 7.86171727578226
70 | 15.7315584392027 | 0.0001 | 3.04279502133862 | 109.069791108056 | 7.82704822397644
71 | 15.9599816487289 | 0.0001 | 3.06474237837286 | 109.516897214339 | 7.8642052264593
72 | 15.704856451242 | 0.0001 | 3.04474210419264 | 109.145219596212 | 7.83112918153362
73 | 15.5382570624066 | 0.0001 | 3.02889810513239 | 111.084801462673 | 7.93465790012539
74 | 15.4655569489629 | 0.0001 | 3.02354849841415 | 110.496768117522 | 7.8544505143097
75 | 15.8806582127796 | 0.0001 | 3.04815502594005 | 109.161117187968 | 7.82285564172074
76 | 16.0161833121095 | 0.0001 | 3.06688852474686 | 109.430631880792 | 7.82435198964003
77 | 15.6582939073008 | 0.0001 | 3.02283104927162 | 108.803929578538 | 7.82973128425766
78 | 15.0873785919131 | 0.0001 | 2.97387013334833 | 109.622737178638 | 7.84787019444335
79 | 15.55741861167 | 0.0001 | 3.01772677172984 | 111.50121805149 | 7.98538517471928
80 | 15.7484325670316 | 0.0001 | 3.05125760449481 | 108.994886085605 | 7.8358045890713
81 | 15.3047282242238 | 0.0001 | 3.01696345082984 | 108.422319980961 | 7.82012074105692
82 | 15.6948753890662 | 0.0001 | 3.03355029375289 | 109.364313935731 | 7.80193253514408
83 | 15.7591919307224 | 0.0001 | 3.0343159074468 | 109.443541672177 | 7.85856403851898
84 | 15.637452715027 | 0.0001 | 3.02520442305996 | 109.8542938269 | 7.89183310008575
85 | 15.395852271255 | 0.0001 | 2.99946590205614 | 108.793636201326 | 7.81461995537192
86 | 15.3001312087698 | 0.0001 | 2.98642063769188 | 108.706381617662 | 7.80204323039873
87 | 15.0091090008124 | 0.0001 | 2.99525076873461 | 108.17255697863 | 7.80665433349683
88 | 15.3955519054287 | 0.0001 | 3.01822176205809 | 110.014424880879 | 7.86972955180723
89 | 15.5081024256744 | 0.0001 | 3.02394761842692 | 108.751448776669 | 7.82268157009882
90 | 15.6578408759007 | 0.0001 | 3.03925812090191 | 112.09883549618 | 7.99767489264133
91 | 14.9970678746215 | 0.0001 | 2.97414287255035 | 108.993256694176 | 7.78585198833067
92 | 15.542554817876 | 0.0001 | 3.00372350450104 | 110.38362696164 | 7.90878225630105
93 | 14.9963628374108 | 0.0001 | 2.9748945732007 | 110.194812781973 | 7.86422222572685
94 | 15.5400248108495 | 0.0001 | 2.99372916274169 | 109.089436908696 | 7.79405554006107
95 | 15.1694773713767 | 0.0001 | 2.98619993977074 | 109.591372759665 | 7.86328218630153
96 | 15.2097082878336 | 0.0001 | 2.98953247013193 | 110.64468777694 | 7.91842743488652
97 | 15.2190196791733 | 0.0001 | 2.98625213881686 | 109.338884980887 | 7.84865158241982
98 | 14.7066756729652 | 0.0001 | 2.93122142847497 | 108.881876984813 | 7.79891812012728
99 | 14.9958411003336 | 0.0001 | 2.94211359108585 | 109.604922531435 | 7.85303175460801
